File: CS/CustomLabelsWithaLargeAmountOfData/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace CustomLabelsWithaLargeAmountOfData
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            // TODO: This line of code loads data into the 'nwindDataSet.Products' table. You can move, or remove it, as needed.
            this.productsTableAdapter.Fill(this.nwindDataSet.Products);

        }

        private void chartControl1_CustomDrawSeriesPoint(object sender, DevExpress.XtraCharts.CustomDrawSeriesPointEventArgs e)
        {
            int index = e.Series.Points.IndexOf(e.SeriesPoint);
            if (index == 0)
            {
                if (e.SeriesPoint.Values[0] <= e.Series.Points[index+1].Values[0])
                    e.LabelText = "";
                return;
            }
            if (index == e.Series.Points.Count - 1)
            {
                if (e.SeriesPoint.Values[0] <= e.Series.Points[index - 1].Values[0])
                    e.LabelText = "";
                return;
            }
            if ((e.SeriesPoint.Values[0] <= e.Series.Points[index - 1].Values[0]) || (e.SeriesPoint.Values[0] <= e.Series.Points[index + 1].Values[0]))
                e.LabelText = "";
            
        }
    }
}

File: VB/CustomLabelsWithaLargeAmountOfData/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Namespace CustomLabelsWithaLargeAmountOfData
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			' TODO: This line of code loads data into the 'nwindDataSet.Products' table. You can move, or remove it, as needed.
			Me.productsTableAdapter.Fill(Me.nwindDataSet.Products)

		End Sub

		Private Sub chartControl1_CustomDrawSeriesPoint(ByVal sender As Object, ByVal e As DevExpress.XtraCharts.CustomDrawSeriesPointEventArgs) Handles chartControl1.CustomDrawSeriesPoint
			Dim index As Integer = e.Series.Points.IndexOf(e.SeriesPoint)
			If index = 0 Then
				If e.SeriesPoint.Values(0) <= e.Series.Points(index+1).Values(0) Then
					e.LabelText = ""
				End If
				Return
			End If
			If index = e.Series.Points.Count - 1 Then
				If e.SeriesPoint.Values(0) <= e.Series.Points(index - 1).Values(0) Then
					e.LabelText = ""
				End If
				Return
			End If
			If (e.SeriesPoint.Values(0) <= e.Series.Points(index - 1).Values(0)) OrElse (e.SeriesPoint.Values(0) <= e.Series.Points(index + 1).Values(0)) Then
				e.LabelText = ""
			End If

		End Sub
	End Class
End Namespace